packages/core-components/src/components/Table/Table.tsx

```diff
@@ -368,14 +377,15 @@ export function Table<T extends object = {}>(props: TableProps<T>) {
     if (typeof data === 'function') {
       return;
     }
-    if (!selectedFilters) {
+    if (!selectedFilters && !search) {
```

Comment:
I think what @benjdlambert meant by a separate state to keep visible data is to swap `tableData` with `visibleData` when there is a search value, so there's probably no need to have a visible data context/provider... I wanted to know if it works by adding an "if" in this effect function that replaces `tableData` with the `visibleData` when a search term is not empty. I tried it [here](https://github.com/backstage/backstage/compare/camilal/filter-catalog-table-data?expand=1), could you take a look if it works for you? If yes, feel free to cherry-pick the changes 🙂

Reply:
Sure, I'll give it a try